tests/datadriven/test_variants.py

```diff
@@ -91,4 +98,41 @@ def testSearchAllVariants(self):
 
     def testVariantSetMetadata(self):
         # TODO get the metadata and test it.
-        pass
+        print("Testing variantset metadata " + self._vcfFileName)
+        keyMap = {}
+        for metadata in self._gaObject.getMetadata():
+            keyMap[metadata.key] = metadata
+
+        # Test version
+        metadata = keyMap["version"]
+        self.assertEqual(metadata.value, self._vcfVersion)
+
+        # Test info
+        for info_i in self._infos.keys():
+            self.assertEqual( keyMap["info."+info_i].type, self._infos[info_i].type )
+            pyvcf_number = self._infos[info_i].num
+            if pyvcf_number == -1:
+                pyvcf_number = "A"
+            elif pyvcf_number == -2:
+                pyvcf_number = "G"
+            elif pyvcf_number is None:
+                pyvcf_number = "."
+            else:
+                pyvcf_number = `pyvcf_number`
+            self.assertEqual( keyMap["info."+info_i].number, pyvcf_number )
+
+        # Test format
+        for format_i in self._formats.keys():
+            self.assertEqual( keyMap["format."+format_i].type, self._formats[format_i].type )
+            pyvcf_number = self._formats[format_i].num
+            if pyvcf_number == -1:
+                pyvcf_number = "A"
+            elif pyvcf_number == -2:
+                pyvcf_number = "G"
+            elif pyvcf_number is None:
+                pyvcf_number = "."
+            else:
+                pyvcf_number = `pyvcf_number`
+            self.assertEqual( keyMap["format."+format_i].number, pyvcf_number )
```

Comment:
No worries Joe, it's just the way we prefer to implement it here.  Here are a few examples that describe this:

https://mkaz.com/2012/10/10/python-string-format/

~p